demos/tests/cases.py

```diff
@@ -68,6 +71,58 @@ def fixed_args(self, source_dir, build_dir):
             *(['-l', str(cpu_extension_path)] if cpu_extension_path.exists() else [])]
 
 
+class NotebookDemo(Demo):
+    """
+    Converts notebook to a Python script with nbconvert. Adds code to make the
+    script work with command line parameters. The `test_code` parameter is
+    added at the end of the script. This should contain the line of code to
+    execute to test the notebook.
+    """
+
+    def __init__(self, name, implementation='jupyter_python', device_keys=None, test_cases=None, test_code=None):
+        super().__init__(name, implementation, device_keys, test_cases)
+        self._exec_name = self._exec_name.replace('_jupyter_python', '')
+        self.test_code = test_code
+
+    def fixed_args(self, source_dir, build_dir):
+        notebook_file = source_dir / self.subdirectory / (self._exec_name + '.ipynb')
+        python_file = notebook_file.with_suffix('.py')
+        python_test_file = python_file.with_name(python_file.stem + '_test.py')
+        subprocess.run([sys.executable, '-m', 'jupyter', 'nbconvert', '--to', 'python', str(notebook_file)])
+
+        original_content = python_file.read_text()
+        original_content = original_content.replace("DOWNLOAD_MODELS = True", "DOWNLOAD_MODELS = False")
+        # Add `test_code` to the test script and make it accept command line arguments
+        content = f"""
+import os
+os.chdir(r'{os.path.dirname(python_file)}')
```

Comment:
```suggestion
os.chdir({os.path.dirname(python_file)!r})
```

Let Python escape any special characters for you. :-)